### PR TITLE
Refactor [v108] iPad Desktop UA to emulate a more recent safari version

### LINKED ISSUE #10005: Slack on iPad says that the browser is deprecated
## Steps to reproduce
Go to https://slack.com/signin

### Expected behavior
Slack works as expected

### Actual behavior
Browser is deprecated

### Device & build information
* Device: iPadOS 15.1
* Build version: Firefox Daylight 96.0 (7403)

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-3756)

### PR DESCRIPTION
**Pull request by Dilatorily**
Update the user agent of Firefox on iPad to a more recent version of Safari.

Fixes #10005.

### CODE REVIEW

**Comment by electricRGB** [comment]
I believe there was a reason why we were keeping our UA at that particular version for desktop. @nbhasin2 or @dnarcese , do you have context?

**Comment by Dilatorily** [comment]
@electricRGB I believe the UA string was hard-coded following https://github.com/mozilla-mobile/firefox-ios/issues/6310#issuecomment-611667720 to fix Reddit and Bank of America.

**Comment by nbhasin2** [comment]
This, in general, is okay but before we do this we should sync up with webcompact team.

**Comment by Inawire** [comment]
FYI: Discourse discussion forms will stop supporting Safari 13 as of January 2023. Looks like they are detecting this UA and presenting warnings with a link to https://meta.discourse.org/t/maintaining-support-for-ios-12-and-ios-13-until-january-2023/224747

View these in Firefox on iOS:
https://discourse.aurelia.io/
https://forums.rockylinux.org/

**Comment by nbhasin2** [comment]
This has been raised with the team and waiting to hear back soon. Don't have a specific update on it at the moment but rest assured I have a reminder to come back to it by end of month.

**Comment by nbhasin2** [comment]
@Dilatorily could you update the UA string to the following 
10_15_4 -> 10_15_7
13.1 -> 16.0

```
Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/16.0 Safari/605.1.15
```

File: Shared/UserAgent.swift [hunk 1]
```diff
@@ -49,7 +49,7 @@ open class UserAgent {
     }
 
     public static func desktopUserAgent() -> String {
-        return "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_4) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/13.1 Safari/605.1.15"
+        return "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/15.3 Safari/605.1.15"
```
Comment (nbhasin2) [inline Shared/UserAgent.swift:52]:
```suggestion
        return "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/16.0 Safari/605.1.15"
```

Will be updating this to the latest iOS 16 safari UA for iPad OS

**Review by nbhasin2 — APPROVED** [review]

**Pull request merged**